lib/mastodon/cli/domains.rb

```diff
@@ -4,7 +4,90 @@
 require_relative 'base'
 
 module Mastodon::CLI
+  # This class is a helper so we can reuse the validation logic in
+  # Admin::Import, which typically expects ActionDispatch::Parameters as an
+  # argument
+  class DomainBlocklistData
+    attr_accessor :original_filename, :path
+
+    def self.from(filename, filepath)
+      import_data = new
+      import_data.original_filename = filename
+      import_data.path = filepath
+      import_data
+    end
+  end
+
+  class DomainBlocklist < Base
+    desc 'import [NAME] [URL]', 'Imports a blocklist from the given URL'
+    def import(name, url)
+      filename = "#{name}.csv"
+      tmpfile = Tempfile.new(filename)
+
+      begin
+        say('Fetching blocklist...')
+        response = HTTP.headers(
+          'Accept' => 'text/csv',
+          'User-Agent' => Mastodon::Version.user_agent
+        ).get(url)
+
+        unless response.code == 200
+          say("Error: Failed to fetch blocklist from: #{url}", :red)
+          exit(1)
+        end
+
+        tmpfile.write(response.body.to_s)
+
+        say('Fetched blocklist!')
+
+        import = Admin::Import.new(data: DomainBlocklistData.from(filename, tmpfile.path))
```

Comment:
This class has some good validation logic inside of it, but ties directly to the http/routing stack, hence needing this workaround.

Reply:
An alternative would be to rework `Admin::Import` to accept a `filename` and `file / filepath` values, which could then work with my approach of using a tmpfile, but that'd be a bunch of extra changes besides just adding the feature to see how people like it.